app/src/main/java/com/cornellappdev/android/eatery/LoginFragment.java

```diff
@@ -0,0 +1,188 @@
+package com.cornellappdev.android.eatery;
+
+
+import android.os.Bundle;
+import android.support.v4.app.Fragment;
+import android.support.v4.app.FragmentTransaction;
+import android.util.Log;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.webkit.WebView;
+import android.webkit.WebViewClient;
+import android.widget.Button;
+import android.widget.CheckBox;
+import android.widget.CompoundButton;
+import android.widget.EditText;
+import android.widget.ProgressBar;
+import android.widget.TextView;
+
+import com.cornellappdev.android.eatery.model.BrbInfoModel;
+import com.cornellappdev.android.eatery.network.GetLoginUtilities;
+import com.cornellappdev.android.eatery.network.JsonUtilities;
+import com.cornellappdev.android.eatery.presenter.AccountPresenter;
+
+public class LoginFragment extends Fragment {
+    private TextView mDescriptionText;
+    private Button mLoginButton;
+    private EditText mNetID;
+    private EditText mPassword;
+    private TextView mPrivacy;
+    private ProgressBar mProgressBar;
+    private CheckBox mSaveInfoCheck;
+    private AccountInfoFragment accountInfoFragment = new AccountInfoFragment();
+
+    private AccountPresenter mAccountPresenter = new AccountPresenter();
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+
+        View rootView = inflater.inflate(R.layout.fragment_login, container, false);
+        getActivity().setTitle("Login");
+
+       if(mAccountPresenter.isLoggedIn()){
+            loadAccountPage(true);
+            return rootView;
+        }
+        mPrivacy = rootView.findViewById(R.id.privacyStatement);
+        mDescriptionText = rootView.findViewById(R.id.descriptionText);
+        mSaveInfoCheck = rootView.findViewById(R.id.saveInfoCheck);
+        mLoginButton = rootView.findViewById(R.id.login);
+        mNetID = rootView.findViewById(R.id.net_id_input);
+        mPassword = rootView.findViewById(R.id.password_input);
+        mProgressBar = rootView.findViewById(R.id.progress_loader);
+        mProgressBar.setVisibility(View.INVISIBLE);
+        mProgressBar.getIndeterminateDrawable().setColorFilter(0xffffffff,
+                android.graphics.PorterDuff.Mode.MULTIPLY);
+        if(mAccountPresenter.isLoggingIn()) {
+            mAccountPresenter.setContext(getContext());
+            String[] loginInfo = mAccountPresenter.readSavedCredentials();
+            if(loginInfo!=null){
+                mNetID.setText(loginInfo[0]);
+                mPassword.setText(loginInfo[1]);
+                mSaveInfoCheck.setChecked(true);
+                mAccountPresenter.setSaveCredentials(true);
+            }
+            loadingGUI();
```

Comment:
does `loadingGUI()` account for when `loginInfo` is null?

Reply:
loadingGUI just displays the animation for loading. There are two cases for this: if the user clicks login (without saving data), navigates to another page and comes back to LoginFragment, then the fragment should still be displaying the loading information even though loginInfo is null (tbh the fragment should already display loadingGUI but this is just insurance).

Secondly, if the user was automatically logged in on app launch and navigates to this page before it finished loading it should display the loadinggui, but also set the text to whatever was stored in the file (because loginInfo is not null, and it autofills the text with whatever your saved password/netid was)

I threw the gist of the above in comments and I know the logic for all of this is kinda convoluted but I couldn't think of a better way to do it